### ISSUE TITLE
Extended types are not working fine....

### ISSUE THREAD

**Issue by xhubbsaxo** [facebook/flow#7407]
If this is a bug, please try reproducing using https://flow.org/try/#0JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcARNnjGQNwBQdMAnmFnAIJwC8cA3nXDjAAJgC44qGFGAA7AOYAaAXAA2yAEZYVAfnGTp8pQF8GzVnABC3DnABkfZcFQBlGMhjBcuuOogQVODJ0JnQB8MiinDy8IuIA5ACMcUb0YT6iVtGxcHEATMn0dAD0RXAA7tAA1qh0yNbqhWZsAJIwWCBRmDj4AHQAogEgWDIwADyUvQDCJJAywzAAKixYo+wAfGv0TXCt7ZldVP2D8+PdMD3T4BBzI0usoxYbjctwkwAWwCrCUMPWACSUYQAeRkKiY7CgUGQTFGuw6mwYBAArjJ8MBrjs2h1LmAABRgYhgVCRACU-EEPxgSKgMnIcPYtGCiJRaIxcIsOPxhOJFjJykp1NpZHZjJCJTgWEh0DgAFo4CBkJUsOgZBAJMNUFgetqeqEsPBZJJ2KJ6dZYVj2DihMIuGQEmQigi0oaYBYTRazeyrSJbfbHYUXZ0XRZCmlkO9Pt9hqII18frSeABtc3tS0za2+h1rAC6qX1PljUZkMY+cd+SZTIA56Z9dqzuYY4slxCgsvliuVcFV6pkmp1uvDpaL9UL8cKQA

In short:

```code
type A = {
  id: string,
  label?: string,
}

type B = A & {
  isStatic?: boolean
}

let a:A = {id: '1'};
let b:B = {id: '2'};

// works fine as expected
a = b;

const ItemAComp = (props:A) => ""
const ItemBComp = (props:B) => ""

// error - makes no sense....
let instA:ItemA = <ItemAComp id="1"/>;
let instB:ItemA = <ItemBComp id="1"/>;

````

Basically that means we are not able to create any custom variation of component with props type A & something, and use it in place where A type is required.

More examples in flow try,

**Comment by goodmind** [comment]
Because `ItemAComp` and `ItemBComp` have different default props?

**Comment by xhubbsaxo** [comment]
@goodmind I don't think so.... no default props in that case at all....

**Comment by vicapow** [comment]
So I *think* this is working as expected because React.Element as defined as an exact type. The equivalent object type would be:

````js
type A = {| id: string, label?: string, |}
type B = A & {| isStatic?: boolean |}
````

The better comparison between the JSX version and the POJO version is more like this:

````js
type A = {| id: string, label?: string, |}
type B = A & {| isStatic?: boolean |}
let a: A = { id: '1' };
function component(props: B) { return props; }
let c: A = component({id : '2'});
````

[try flow](https://flow.org/try/#0C4TwDgpgBAglC8UDeAfKBLAJgLigZ2ACd0A7AcwBooAbAQwCMJqB+XA48qlAXwChRIUAEIJYUAGTI06PAGVgtYOgDGrKPQD2G6hFokoPXrx3AotXHERIMOKAHIAjHajcA3EYBmAVxLKlG-WUNAFswAIgSYAAKMEINMDxcIQBKZChCCGAvQn1Y+LxXF2NMqGULUSDQ8MiopCwoXDsAJjtuZPcjIyA)

**Comment by vicapow** [comment]
Closing but please reopen if I missed anything.

**Comment by xhubbsaxo** [comment]
@vicapow your explanation is quite clear: "This is working as expected" but, I would like to get to the bottom of problem because we are starting having similar problems in multiple places related to list based components.

This is very basic real world example:

```
<List>
    <MyCustomItem1 id={1} myCustomProp1={1} />
    <MyCustomItem2 id={1} nyCustomProp2={2} />
</List>
```
From List component point of view:
- children must have id prop
- it can have anything else custom (we don't care about it, and we should not block it)

How it's possible using current flow version to declare type for List children prop?
